### PR TITLE
easy_labs

### PR DESCRIPTION
**Pull request by yonicd**

### CODE REVIEW

**Comment by timchurches** [comment]
easy_labs() applies the Species column label from iris1 where it should apply the  Species column label form iris2 (which is where the Species data is coming from) in this example:

```
iris1 <- iris
labelled::var_label(iris1$Sepal.Length) <- "Length of sepal"
labelled::var_label(iris1$Sepal.Width) <- "Width of sepal"
labelled::var_label(iris2$Species) <- "Sub-genera"
iris2 <- iris1
iris2$Species <- as.character(iris2$Species)
iris2[iris2$Species=="setosa",]$Species <- "rainbow"

ggplot(data=iris1, aes(x=Sepal.Length,y=Sepal.Width)) + geom_point(data=iris2, aes(colour=Species)) + easy_labs()
```

The way to address this is to looking at the aesthetic mappings for the main dataframe, and then get the column labels for each of this mapping, and apply the col labels to the appropriate item in p$labels. Then repeat that for each of the additional layers created by germs, if the data element for that layer is not empty (which implies an additional data frame was supplied for that layer).

Secondary axes are not handled either.

**Comment by yonicd** [comment]
not sure what you are seeing. the label from iris2 is being used

image

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@
 .Ruserdata
 ggeasy.Rproj
 inst/doc
+NAMESPACE
```
Comment (jonocarroll) [inline .gitignore]: I suspect you don't mean for this to be ignored.

File: .gitignore [hunk 2]
```diff
@@ -4,3 +4,5 @@
...
+labs_utils.R
```
Comment (jonocarroll) [inline .gitignore]: Accidentally ignored?

File: .gitignore (hunk 1 above)
Comment (yonicd) [reply .gitignore]: it was pushed anyways... ignore only works on nontracked files

File: .gitignore [hunk 3]
```diff
@@ -4,3 +4,5 @@
 .Ruserdata
 ggeasy.Rproj
 inst/doc
```
Comment (jonocarroll) [inline .gitignore]: I'm not sure how long this has been there (prior to this PR it seems), but FYI once I build a `pkgdown` site (soon) this will get in the way.

File: .gitignore (hunk 2 above)
Comment (yonicd) [reply .gitignore]: `lab_utils.R` is the +gg variation of `easy_labs`...i'll keep it for bookkeeping

File: R/axis.R [hunk 4]
```diff
@@ -32,8 +32,8 @@ easy_rotate_x_labels <- function(angle = 90, side = c("left", "middle", "right")
     angle
   } else if (is.character(angle)) {
     switch(angle,
-      startattop = -90,
-      startatbottom = 90
+           startattop = -90,
```
Comment (jonocarroll) [inline R/axis.R]: Could you please revert this file if there are only unrelated whitespace changes?

File: DESCRIPTION [hunk 5]
```diff
@@ -3,7 +3,8 @@ Title: Easy Access To `ggplot2` Commands
 Version: 0.1.0
 Authors@R: c(
  person("Jonathan", "Carroll", email = "rpkg@jcarroll.com.au", role = c("aut", "cre"), comment = c(ORCID = "0000-0002-1404-5264")),
- person("Alicia", "Schep", email = "aschep@gmail.com", role = c("aut"), comment = c(ORCID = "0000-0002-3915-0618"))
+ person("Alicia", "Schep", email = "aschep@gmail.com", role = c("aut"), comment = c(ORCID = "0000-0002-3915-0618")),
+ person("Jonathan", "Sidi", email = "yonicd@gmail.com", role = c("aut"))
```
[the remark this answers is not in the document]
Comment (jonocarroll) [reply DESCRIPTION]: Plus it generates a fancy-shmancy icon for each author on the CRAN landing page.

File: R/labs.R [hunk 6]
```diff
@@ -0,0 +1,114 @@
+#' @title Easily add ggplot labels using label attribtute of data.frame column
+#' @description Applies same logic as \code{\link[ggplot2]{labs}} but uses as default
+#' the column label attribute if present as the variable label in the plot.
+#' @param ... A list of new name-value pairs. The name should either be an aesthetic, 
+#' or one of "title", "subtitle", or "caption"
+#' @return NULL
```
[the remark this answers is not in the document]
Comment (yonicd) [reply R/labs.R]: i went off of `ggplot2::labs()`... which has NULL in the return

File: .gitignore (hunk 3 above)
Comment (AliciaSchep) [reply .gitignore]: Doesn't pkgdown usually build in docs folder in main dir?

Comment (yonicd) [reply .gitignore]: i think when we get to that bridge we can pass it and remove it from the gitignore.

File: .Rbuildignore [hunk 7]
```diff
@@ -3,3 +3,6 @@
 
 ^README\.Rmd$
 ^README-.*\.png$
+^.*\.yml$
+
+labs_utils.R
```
[the remark this answers is not in the document]
Comment (yonicd) [reply .Rbuildignore]: - im using sinewconfig on my side, that needs to be ignored by pkg. when is a yml needed in a pkg, they are always in buildignore arent they?

  - the R file is still on my local checkout, and needs to be ignored.

File: inst/doc/shortcuts.Rmd [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+## Plot Labels
+
+```{r, fig.width = 8, fig.height = 8}
+iris_labs <- iris
+
+lbl <- c('Sepal Length','Sepal Width','Petal Length','Petal Width','Flower Species')
+
+var_label(iris_labs) <- split(lbl,names(iris_labs))
+
+p <-
```
Comment (jonocarroll) [inline inst/doc/shortcuts.Rmd]: While this perhaps lines up the following lines better, the end of line after `<-` is very odd.

File: inst/doc/shortcuts.Rmd [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+plot_grid(p1, p2, p3, p4, nrow = 2)
+```
+
+## Plot Labels
+
+```{r, fig.width = 8, fig.height = 8}
+iris_labs <- iris
+
+lbl <- c('Sepal Length','Sepal Width','Petal Length','Petal Width','Flower Species')
```
Comment (jonocarroll) [inline inst/doc/shortcuts.Rmd]: I'd like to maintain spaces after commas throughout.

File: inst/doc/shortcuts.Rmd [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+
+var_label(iris_labs) <- split(lbl,names(iris_labs))
+
+p <- 
+     ggplot(iris_labs,aes(x=Sepal.Length,y=Sepal.Width))+
+     geom_line(aes(colour=Species)) +
+     theme(text=element_text(size=7))
```
Comment (jonocarroll) [inline inst/doc/shortcuts.Rmd]: `size=7` makes these small for me. In general I'd like to avoid adding any commands which don't highlight the difference between the 'easy' way and the 'hard' way, even if they look nicer.

File: inst/doc/shortcuts.Rmd [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+#complex usage: 
+#  adding layer level data with same column names but different labels
+
+iris_labs_2 <- iris_labs
+var_label(iris_labs_2$Species) <- "Sub-genera"
+
+p4 <- p + 
+  geom_point(data=iris_labs_2, aes(fill=Species),shape=24) +
+  facet_wrap(~Species,labeller = label_both) +
```
Comment (jonocarroll) [inline inst/doc/shortcuts.Rmd]: While the `labeller` option makes for a nicer plot, I don't want to add distraction from the distinction we're making.

File: R/legend.R [hunk 12]
```diff
@@ -46,8 +46,8 @@ easy_remove_legend <- function(..., teach = FALSE) {
   } else {
     inputs <- lapply(vars, function(x) FALSE)
     names(inputs) <- vars
-    
-    if (teach){
+
+    if (teach) {
```
Comment (jonocarroll) [inline R/legend.R]: Please be careful not to modify (even trivially) unrelated files, simply because it clutters the diff.

File: R/labs.R [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+  }
+  
+  p
+  
+}
+
+#' @export
+ggplot_add.easy_labs <- function (p, object, objectname) {
+  
+    p <- easy_update_labels(object,p)
```
Comment (jonocarroll) [inline R/labs.R]: From a style perspective, is there a reason to modify `p` in place here rather than just return `easy_update_labels(object, p)`?

File: R/labs.R [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+    for(nm in names(man_labs)){
+      args[[nm]] <- man_labs[[nm]]
+      p$labels[[nm]] <- man_labs[[nm]]
+    }
+  
+  plot_names <- unique(c(names(p$data),
+                         unlist(lapply(p$layers,function(x) names(x$data)))))
+  
+  for(i in names(args)){
+    if(p$labels[[i]]%in%plot_names) #making sure only replacing default col names
```
Comment (jonocarroll) [inline R/labs.R]: Spaces become particularly useful on lines like this one.

File: R/labs.R [hunk 15]
```diff
@@ -0,0 +1,103 @@
+#' @title Easily add ggplot labels using label attribtute of `data.frame` column
+#' @description Applies same logic as \code{\link[ggplot2]{labs}} but uses as default
+#' the column label attribute if present as the variable label in the plot.
+#' @param ... A list of new name-value pairs. The name should either be an aesthetic, 
+#' or one of "title", "subtitle", or "caption"
+#' @return NULL
+#' @examples 
+#' 
+#' require(ggplot2)
```
Comment (jonocarroll) [inline R/labs.R]: `library()` rather than `require()`.

File: R/labs.R [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+#'   
+#' p
+#' p + easy_labs()
+#' p + easy_labs(subtitle='mysubtitle', x='x axis label')
+#'  
+#' @rdname easy_labs
+#' @export 
+easy_labs <- function(...){
+
+  man_labs <- labs(list(...))
```
Comment (jonocarroll) [inline R/labs.R]: Do we need to explicitly namespace prefix or explicitly `importFrom`? `ggplot2::labs()`

File: R/labs.R (hunk 6 above)
[the remark this answers is not in the document]
Comment (jonocarroll) [reply R/labs.R]: But this *doesn't* return `NULL`. It returns a structure.

File: R/labs.R [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+  
+  structure(man_labs, class = "easy_labs")
+  
+}
+
+easy_update_labels <- function(p,man_labs){
+
+  root_dat_labs <- sapply(p$data,attr,which='label')
+  root_dat_labs <- root_dat_labs[!sapply(root_dat_labs,is.null)]
```
Comment (jonocarroll) [inline R/labs.R]: Consider `non_null_root_dat_labs <- Filter(Negate(is.null), root_dat_labs)` as an alternative, which may be easier to read.

File: R/labs.R [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+    }
+
+    new_lab
+    
+  })
+  
+  
+  layers_map <- lapply(p$layers,function(x){
+    
+    l_dat_labs <- sapply(x$data,attr,which='label')
```
Comment (jonocarroll) [inline R/labs.R]: This feels like a lot of duplicated code but the one line that's different makes it not quite modular.
